bls/src/test/java/org/dashj/bls/AdvancedTest.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2022-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package org.dashj.bls;
+
+import org.dashj.bls.Utils.ByteVector;
+import org.dashj.bls.Utils.ByteVectorList;
+import org.dashj.bls.Utils.G1ElementList;
+import org.dashj.bls.Utils.G2ElementList;
+import org.dashj.bls.Utils.PrivateKeyList;
+import org.dashj.bls.Utils.Util;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+public class AdvancedTest extends BaseTest {
+
+    @Test
+    public void shouldAggregateWithMultipleLevelsDegenerate() {
```

Comment:
This is the first of many tests that are included in the BLS library.  All of the tests, except a few, were translated into Java to test over 90% of the functions in the BLS classes.